fe/fe-core/src/main/java/org/apache/doris/statistics/BaseAnalysisTask.java

```diff
@@ -165,6 +164,7 @@ protected void init(AnalysisInfo info) {
             Preconditions.checkArgument(!StatisticsUtil.isUnsupportedType(col.getType()),
                     String.format("Column with type %s is not supported", col.getType().toString()));
         }
+        tableSample = getTableSample();
```

Comment:
Why move it here? This line should before line 155, otherwise, external table will failed to get table row count while sampling.

Reply:
We must make sure tbl field has been init before invoking getTableSample